tests/spec_tests/query_tests/type_conversions.yml

```diff
@@ -981,6 +984,24 @@ tests:
       - {'': {'type': 'minKey', 'asString': null}}
       - {'': {'type': 'maxKey', 'asString': null}}
 
+    # in mongodb 8.3.0, many of the previously unsupported casts to string became valid. This test
```

Comment:
I considered breaking the invalid types dataset up entirely into the individual types and adding two tests for each -- one with a max version of 8.2 and one with a min version of 8.3.

For now, I decided it would be easier to just leave this as is, but happy to bikeshed on this if anyone feels strongly.